File: healthcare/sales_invoice.py
import frappe
def before_save(doc,event):
    doc.additional_discount_percentage = 5
    
    total = doc.rounded_total
    insurance_percentage = doc.custom_insurance_percentage or 0  
    insurance_amount = (insurance_percentage / 100) * total
    net_payable = total - insurance_amount
    
    # if net_payable < 0:
    #     frappe.throw("Insurance coverage exceeds total amount.")

    doc.net_total = net_payable
    doc.rounded_total = round(net_payable)  
    doc.grand_total = net_payable
    doc.outstanding_amount = net_payable
    

File: healthcare/public/js/sales_invoice.js
// frappe.ui.form.on('Sales Invoice', {
    

//     validate: function (frm) {

//         let insurance_amount = (frm.doc.custom_insurance_percentage / 100) * frm.doc.total;
//         frm.doc.grand_total = frm.doc.grand_total - insurance_amount;

//         // Show a message if insurance covers everything
//         if (frm.doc.grand_total <= 0) {
//             frappe.msgprint(__('Insurance coverage exceeds or matches total amount.'));
//             frm.doc.grand_total = 0;
//         }
//     }



// });
